packages/truffle-core/lib/console.js

```diff
@@ -1,248 +1,225 @@
-  }
+    }
 
-  // Much of the following code is from here, though spruced up:
-  // https://github.com/nfcampos/await-outside
-
-  /*
-  - allow whitespace before everything else
-  - optionally capture `var|let|const <varname> = `
-    - varname only matches if it starts with a-Z or _ or $
-      and if contains only those chars or numbers
-    - this is overly restrictive but is easier to maintain
-  - capture `await <anything that follows it>`
-  */
-  let includesAwait = /^\s*((?:(?:var|const|let)\s+)?[a-zA-Z_$][0-9a-zA-Z_$]*\s*=\s*)?(\(?\s*await[\s\S]*)/;
-
-  var match = cmd.match(includesAwait);
-  var source = cmd;
-  var assignment = null;
-
-  // If our code includes an await, add special processing to ensure it's evaluated properly.
-  if (match) {
-    var assign = match[1];
-    var expression = match[2];
-
-    var RESULT = "__await_outside_result";
-
-    // Wrap the await inside an async function.
-    // Strange indentation keeps column offset correct in stack traces
-    source = `(async function() { try { ${
-      assign ? `global.${RESULT} =` : "return"
-    } (
-${expression.trim()}
-); } catch(e) { global.ERROR = e; throw e; } }())`;
-
-    assignment = assign
-      ? `${assign.trim()} global.${RESULT}; void delete global.${RESULT};`
-      : null;
-  }
+    // Much of the following code is from here, though spruced up:
+    // https://github.com/nfcampos/await-outside
+
+    /*
+    - allow whitespace before everything else
+    - optionally capture `var|let|const <varname> = `
+      - varname only matches if it starts with a-Z or _ or $
+        and if contains only those chars or numbers
+      - this is overly restrictive but is easier to maintain
+    - capture `await <anything that follows it>`
+    */
+    let includesAwait = /^\s*((?:(?:var|const|let)\s+)?[a-zA-Z_$][0-9a-zA-Z_$]*\s*=\s*)?(\(?\s*await[\s\S]*)/;
+
+    var match = cmd.match(includesAwait);
+    var source = cmd;
+    var assignment = null;
+
+    // If our code includes an await, add special processing to ensure it's evaluated properly.
+    if (match) {
+      var assign = match[1];
+      var expression = match[2];
+
+      var RESULT = "__await_outside_result";
+
+      // Wrap the await inside an async function.
+      // Strange indentation keeps column offset correct in stack traces
+      source = `(async function() { try { ${
+        assign ? `global.${RESULT} =` : "return"
+      } (
+  ${expression.trim()}
+  ); } catch(e) { global.ERROR = e; throw e; } }())`;
+
+      assignment = assign
+        ? `${assign.trim()} global.${RESULT}; void delete global.${RESULT};`
+        : null;
+    }
 
-  var runScript = function(s) {
-    const options = {
-      displayErrors: true,
-      breakOnSigint: true,
-      filename: filename
+    const runScript = script => {
+      const options = {
+        displayErrors: true,
+        breakOnSigint: true,
+        filename: filename
+      };
+      return script.runInContext(context, options);
```

Comment:
Are you sure this works? I don't think a function from inside `script` is accessible inside the arrow function because of the way context works in arrow functions.

Reply:
Your concern is about `script` and accessing it's functions? If script has a function on it then it should be fine. The thing to worry about would be the context. I think `context` should be fine here?

Reply:
My concern is that you are passing script as an object in the arrow function parameter, right? And `runInContext()` is an object method of script's, right? That should return undefined if I'm not missing something. Since the context of the arrow function is inherited from the execution context but the arrow function doesn't bind context/this I think there's an issue with making this an arrow function.